src/App.js

```diff
@@ -297,55 +337,24 @@ class FeatureLoader extends React.Component<Props, State> {
   }
 
 
-  featureId(props: Props = this.props): ?string {
-    const routeInformation = this.routeInformation(props);
-    return routeInformation ? routeInformation.featureId : null;
-  }
-
-
-  routeInformation(props: Props = this.props): ?RouteInformation {
-    const location = props.location;
-    const allowedResourceNames = ['nodes', 'categories', 'search'];
-    const match = location.pathname.match(/(?:\/beta)?\/?(?:(-?\w+)(?:\/([-\w\d]+)(?:\/([-\w\d]+)(?:\/([-\w\d]+))?)?)?)?/i);
-    if (match) {
-      if (match[1] && !includes(allowedResourceNames, match[1])) return null;
-      return {
-        featureId: match[1] === 'nodes' ? match[2] : null,
-        equipmentInfoId: (match[1] === 'nodes' && match[3] === 'equipment') ? match[4] : null,
-        category: match[1] === 'categories' ? match[2] : null,
-        searchQuery: match[1] === 'search' ? parseQueryParams(location.search).q : null,
-        isEditMode: (match[3] === 'edit'),
-      };
-    }
-    return null;
-  }
-
-
-  fetchFeature(props: Props): void {
-    const id = this.featureId(props);
-    if (!id) {
-      this.setState({ feature: null });
+  fetchFeature(featureId: ?string): void {
+    if (!featureId) {
+      this.setState({ feature: null, featureId });
       return;
     }
-    this.setState({ fetching: true });
-    const isWheelmap = isWheelmapFeatureId(id);
+    this.setState({ fetching: true, featureId });
+    const isWheelmap = isWheelmapFeatureId(featureId);
     if (isWheelmap) {
-      this.setState({ feature: wheelmapLightweightFeatureCache.getCachedFeature(id) });
+      this.setState({ feature: wheelmapLightweightFeatureCache.getCachedFeature(featureId) });
     }
     const cache = isWheelmap ? wheelmapFeatureCache : accessibilityCloudFeatureCache;
-    cache.getFeature(id).then((feature: AccessibilityCloudFeature | WheelmapFeature) => {
+    cache.getFeature(featureId).then((feature: AccessibilityCloudFeature | WheelmapFeature) => {
       if (!feature) return;
-      const currentlyShownId = this.featureId(this.props);
-      const idProperties = [
-        typeof feature.id === 'number' && feature.id,
-        typeof feature._id === 'string' && feature._id,
-        feature.properties && typeof feature.properties.id === 'number' && feature.properties.id,
-        feature.properties && typeof feature.properties._id === 'string' && feature.properties._id
-      ];
-      const fetchedId = String(idProperties.filter(Boolean)[0]);
+      const currentlyShownId = getFeatureId(this.props);
```

Comment:
-2: Should this not be `getFeatureIdFromProps`?

Reply:
Whoops! Yep!